### PR TITLE
Lesson 6: integers and booleans

### PR DESCRIPTION
**Pull request by dwightguth**
I fix a few minor issues in previous lessons, and then include the text of lesson 6, which discusses integers and booleans. I chose not to go into great detail about all the various integer and boolean operations, instead referring users to the domains.md file which is now relatively documented. Please let me know if you think there is more concrete information that ought to be present within the text of the lesson itself.

### CODE REVIEW

File: k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md [hunk 1]
```diff
@@ -1 +1,173 @@
...
+                | isBlueOrYellow(Fruit) [function]
+
+  rule isYellow(Banana()) => true
+  rule isYellow(Blueberry()) => false
+
+  rule isBlueOrYellow(F:Fruit) => isBlue(F) orBool isYellow(F)
+endmodule
+```
+
+In the above example, Boolean inclusive or is performed via the `orBool`
```
Comment (ehildenb) [inline k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md]:
```suggestion
In the above example, Boolean **inclusive or** is performed via the `orBool`
```

File: k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md [hunk 2]
```diff
@@ -1 +1,173 @@
...
+syntax of K does not (generally) conflict with the syntax of any other
+programming language, which would make it harder to define that programming
+language in K.
+
+### Exercise
+
+Write a function `isBlueAndNotYellow` which computes the appropriate Boolean
+expression. If you are unsure what the appropriate syntax is to use, you
+can refer to the `BOOL` module in
+[domains.md](../../../include/kframework/builtin/domains.md).
```
Comment (ehildenb) [inline k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md]:
```suggestion
Write a function `isBlueAndNotYellow` which computes the appropriate Boolean
expression. If you are unsure what the appropriate syntax is to use, you
can refer to the `BOOL` module in
[domains.md](../../../include/kframework/builtin/domains.md). Write a test for
this function and confirm that running it with the definition returns the correct result.
```

File: k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md [hunk 3]
```diff
@@ -1 +1,173 @@
...
+Note the encapsulation of syntax: the `LESSON-06-C-SYNTAX` module contains
+exactly the syntax of our boolean expressions, and no more, whereas any other
+syntax needed to implement those functaions is in the `LESSON-06-C` module
+instead.
+
+#### Exercise
+
+Add an "implies" function to the above Boolean expression calculator, using the
+`->` symbol to represent implication. You can look up K's builtin "implies"
+function in the `BOOL` module in `domains.md`.
```
Comment (ehildenb) [inline k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md:131]:
```suggestion
function in the `BOOL` module in `domains.md`. Test that this works by using `krun` to evaluate the expressions `true -> true`, `true -> false`, `false -> true`, and `false -> false`.
```

File: k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md [hunk 4]
```diff
@@ -1 +1,173 @@
...
+refer to `domains.md`. Note again how we append `Int` to the end of most of the
+integer operations to ensure they do not collide with the syntax of other
+programming languages.
+
+## Exercises
+
+1. Extend your solution from lesson 4, problem 2 to implement the rules
+that define the behavior of addition, subtraction, multiplication, and
+division. Do not worry about the case when the user tries to divide by zero
+at this time. Use `/Int` to implement division. Test your new calculator
```
Comment (ehildenb) [inline k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md:162]:
```suggestion
at this time (ie. use `/Int` to implement division). Test your new calculator
```

Comment (dwightguth) [reply k-distribution/k-tutorial/1_basic/06_ints_and_bools/README.md:162]: i.e. is not appropriate here.

**Review by ehildenb — APPROVED** [review]

**Pull request merged**